In [3]:
# Importing necessary libraries
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error

# Load your dataset (replace 'data.csv' with your actual dataset)

data = pd.read_excel('C:/Users/USER/Desktop/Pyth Cleaning/Even.xlsx')

data.head()



Customer ID First Name  Last Name      Brand         Product       City  \
0         1295       Bill      Smith       Nike   Magista Shoes     London   
1         1298    Kennedi      Singh     Armani   V-Neck Tshirt     Madrid   
2         1301     Harley      Fritz  Nike Inc.  Football Shirt    Tampa     
3         1304       Nyla      Novak    Nike UK         Air Max  Tokyo       
4         1307      David  Rasmussen       Puma         Air Max  Tokyo       

  Country  Fulfillment Percent  Price  Units Sold  Total Sales  
0      Uk                    0  12279        1200     14734800  
1   Spain                    0  20462        1000     20462000  
2     Usa                   30  12276        1000     12276000  
3   Japan                   30   8160         850      6936000  
4   Japan                    0   5467         900      4920300

In [4]:
# Assuming 'total_sales' is the column you want to predict
X = data[['Price', 'Units Sold']]  # Features: price and units sold
y = data['Total Sales']  # Target variable: total sales

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Initialize the Random Forest regressor
rf_regressor = RandomForestRegressor(n_estimators=100, random_state=42)




In [5]:
# Train the regressor on the training data
rf_regressor.fit(X_train, y_train)

# Make predictions on the testing data
predictions = rf_regressor.predict(X_test)

print('predictions')

predictions


In [6]:
# Evaluate the model (using Mean Squared Error as an example)
mse = mean_squared_error(y_test, predictions)
print("Mean Squared Error:", mse)

Mean Squared Error: 5358179870818.46


The Mean Squared Error (MSE) value you provided, 5,358,179,870,818.46, is a measure of the average squared difference between the actual values and the predicted values of the target variable. In this case, since the units of the MSE are squared units of the target variable, the specific units depend on the units of your target variable.

Here's what this MSE value implies:

1. **Magnitude**: The magnitude of the MSE is quite large, indicating that, on average, the squared differences between the actual and predicted values are substantial.

2. **Interpretation**: Since MSE is a squared error metric, its interpretation can be less intuitive. To gain a more interpretable understanding, you might consider taking the square root of the MSE to obtain the Root Mean Squared Error (RMSE), which is in the same units as the target variable. This would provide a clearer understanding of the average error in the original units of the target variable.

3. **Model Performance**: Without a context for the units of the target variable, it's challenging to assess the model's performance solely based on the MSE value. However, in general, lower MSE values indicate better model performance, while higher MSE values suggest poorer performance.

4. **Comparison**: To assess whether this MSE value is good or bad, it's helpful to compare it to other models or to a baseline. You might compare it to the MSE of a simple baseline model (e.g., predicting the mean of the target variable for all samples) or to the MSE of other regression models trained on the same data.

5. **Further Analysis**: Depending on the context and requirements of your problem, you may want to investigate further to understand why the model's performance resulted in such a high MSE. This could involve examining the data, feature engineering, trying different algorithms or hyperparameters, or collecting additional data if necessary.

In summary, the MSE value of 5,358,179,870,818.46 indicates the average squared difference between the actual and predicted values of the target variable, but its interpretation and assessment of model performance depend on the context and units of the target variable.